programs/patient-program-base.service.js

```diff
@@ -21,74 +22,93 @@ function getAllProgramsConfig () {
-
-    return programVisits.getPatientVisitTypes(patientUuid, programUuid,
-      enrollmentUuid, intendedVisitLocationUuid || '', clone , initialVisit);
+function getPatientProgramEnrollmentVisits(
+  patientUuid, programUuid, enrollmentUuid,
+  intendedVisitLocationUuid, initialVisit
+) {
+  const clone = getAllProgramsConfig();
 
-
- 
+  return programVisitTypes.getPatientVisitTypes(
+    patientUuid, programUuid,
+    enrollmentUuid, intendedVisitLocationUuid || '', clone, initialVisit
+  );
 }
 
 function validateEnrollmentOptions (patient) {
-  var clone = getAllProgramsConfig();
+  const clone = getAllProgramsConfig();
   return programValidationService.validateEnrollmentOptions(patient, clone);
 }
 
-function getPatientProgramVisits(patientUuid, programUuid, enrollment, locationUuid){
 
+function getPatientProgramVisits(patientUuid, programUuid, enrollment, locationUuid) {
   return new Promise(function (success, error) {
-
-     let initialVisit = false;
-
-      if (programUuid === '781d897a-1359-11df-a1f1-0026b9348838') {
-        // only count initial visits for PMTC
-        initialEncounters.getPatientInitialEncounters(patientUuid)
-        .then(function (result) {
-
-          console.log('getPatientInitialEncounters', result);
-
-          let initialVisits
-
-          if(result.length > 0){
-            if(result[0].initial_encounters > 0){
+    let initialVisit = false;
+
+    if (programUuid === '781d897a-1359-11df-a1f1-0026b9348838') {
+      // only count initial visits for PMTCT
+      initialEncounters.getPatientInitialEncounters(patientUuid)
+        .then((result) => {
+          if (result.length > 0) {
+            if (result[0].initial_encounters > 0) {
               initialVisit = true;
-            }else{
+            } else {
               initialVisit = false;
-            }
-
-          }else{
+            }   
+          } else {
             initialVisit = false;
           }
-
-            getPatientProgramEnrollmentVisits(patientUuid, programUuid, enrollment, locationUuid, 
-                initialVisit)
-            .then(function (programVisits) {
-                success(programVisits);
-            })
-            .catch(function (error) {
-                error(Boom.badImplementation('An internal error occurred'));
+          
+          getPatientProgramEnrollmentVisits(
+            patientUuid, programUuid, enrollment, locationUuid, initialVisit
+          )
+            .then((programVisits) => {
+              success(programVisits);
             })
-
+            .catch((err) => {
+              error(Boom.badImplementation('An internal error occurred'));
+            });
+        })
+        .catch((err) => {
+          Error('Error fetching initial encounters: ', err);
         });
+    } else if (programUuid === 'c19aec66-1a40-4588-9b03-b6be55a8dd1d') {
+      // fetch initial PrEP encounters
+      const initialPrEPEncounter = '00ee2fd6-9c95-4ffc-ab31-6b1ce2dede4d';
```

Comment:
move logic to determine whether this is an initial or return prep visit to the scope builder. Use isAmpathFirstVisit scope member for reference.